dataverse-webapp/src/main/java/edu/harvard/iq/dataverse/DatasetPage.java

```diff
@@ -265,6 +266,14 @@ public boolean canPublishDataverse() {
         return permissionsWrapper.canIssuePublishDataverseCommand(dataset.getOwner());
     }
 
+    public boolean latestDatasetHasFiles(){
+        return !dataset.getLatestVersion().getFileMetadatas().isEmpty();
```

Comment:
Question: are we sure that there will be no null in this chain?

Reply:
Yeah, I think so. Dataset always needs at least one DatasetVersion so latestVersion is correct. When creating DatasetVersion the fileMetadatas field will have empty list. So all should be fine.